### ISSUE TITLE
Backport useful features from "trolcommander"

### ISSUE THREAD

**Issue by magicgoose** [mucommander/mucommander#18]
https://github.com/trol73/mucommander

For example, it has a very useful feature — browsing Android devices via ADB interface.  
And there are more of them.

**Comment by ahadas** [comment]
Yeah, that's the plan
But need to release 0.9.1, based on current master branch, first.

**Comment by metim** [comment]
Thanks for the wonderful program!
I would like to point some other useful features which would be great to have in muCommander:
- addition of different colors to files based on extentions,
- deleting to Trash,
- search files,
- bug fixes (fixed status pane - long filenames will go to the second line, fix copy dialog to show whole copy status correctly, etc.).

**Comment by ahadas** [comment]
Hi @metim ,
Colors to files based on extensions - sounds interesting, could you open a separate issue on that? I wonder if the criteria should be file extension or other properties of the file, like if it is executable.
Deleting to trash - that already exists. What operating system do you use?
Search files - sure.
Bugs - could you please file issues on them?

**Comment by metim** [comment]
Hi, @ahadas,
Thanks for the quick answer! It is great that developers can answer so quickly and the project is alive.
I will create new issues. The point I posted they here is that they have been already implemented in the fork so it is possible to take them from there. 
About bugs: I mentioned only what I have encountered so far but there may be other issues already fixed in the fork (but there may be new bugs there as well).
About Trash: It doesn’t work in Windows system - there is no option delete to Trash.

**Comment by magicgoose** [comment]
@metim maybe it's because it's called Recycle Bin in Windows? 😆

**Comment by yashpalgoyal1304** [comment]
thanks @author to open this issue (:

i came here from the mucommander's [wikip page](https://en.wikipedia.org/wiki/MuCommander#Forks), and i found mucommander while searching for FOSS alternate to totalcommander ([src](https://en.wikipedia.org/wiki/Comparison_of_file_managers#Cross-platform_file_managers))

**Comment by yashpalgoyal1304** [comment]
hi @ahadas any updates on this?

**Comment by rokejulianlockhart** [comment]
I believe that the actual features desired should be split each into separate issues and merely linked to here. This issue is probably too vague to be usefully actionable.

**Comment by yashpalgoyal1304** [comment]
My main one is the integrated terminal - which is not available in any maintained FOSS GUI file explorer for windows to my knowledge.

**Comment by ahadas** [comment]
@yashpalgoyal1304 an integrated terminal was contributed by @pskowronek in v1.1.0

**Comment by yashpalgoyal1304** [comment]
> in v1.1.0

i was thinking same, but couldn't decide what version of java my system has

> A Java Runtime Environment (JRE) 11 or later is required to run muCommander.
Java 17 is recommended, you can download it from https://adoptium.net/temurin.

**Comment by ahadas** [comment]
@yashpalgoyal1304 how did you install it? if you use the `msi` installation, it comes with bundled JRE so you don't have to worry about it

**Comment by rokejulianlockhart** [comment]
@yashpalgoyal1304, restoration of a registry backup upon uninstallation could corrupt the registry if the user or other software has modified it. Surely that is obvious?

**Comment by ahadas** [comment]
> Unless this software uses Qt, I doubt that achieving something similar shall be easy unless the developer merely integrates a terminal via something like a git submodule into the source?

we use [jediterm](https://github.com/JetBrains/jediterm) - you can check out the dependencies we use [here](https://github.com/mucommander/mucommander/blob/master/jetbrains-jediterm/build.gradle)

**Comment by yashpalgoyal1304** [comment]
> restoration of a registry backup upon uninstallation could corrupt the registry if the user or other software has modified it.

yes, that's why i put the revo uninstaller method before, and also said "without installing any other software in between".



> how did you install it?

oh, i picked the .zip one for the latest version - will pick the .msi one now.
('have been transitioning to non-installs lately - be via standalones, or via msys2 pacman. As managing the C: drives becoming nasty and windows giving me virtual headaches.)
